xla/service/gpu/gpu_windowed_einsum_handler.cc

```diff
@@ -673,66 +638,142 @@ class WindowedEinsumVisitor : public DfsHloRewriteVisitor {
   absl::Status HandleDot(HloInstruction* dot) override {
     CHECK_EQ(dot->opcode(), HloOpcode::kDot);
     HloComputation* comp = dot->parent();
-    // Rewrites a allgather-dot pattern that shares the same operand
-    // with a windowed einsum loop to consume the output of the loop
-    // and remove the all-gather.
     // Now that we have processed all loops, we can check if there are any
     // allgather-dot pattern that we can optimize. We'd want to transform:
     //                       input
     //                       /    |
-    //                      /     |
-    //                     AG    windowed loop
-    //                     /
-    //                    /
-    //                   dot
+    //                dequantize  |
```

Comment:
Is dequantize optional or always needed to match?

Reply:
The dequantization step is optional. When the data type is FP8 and there’s a dequantization, it will be shifted as described in the comment. This doesn't interfere with other cases.